### ISSUE TITLE
Running detekt with type resolution emits error messages in presence of "kotlin-parcelize" plugin

### ISSUE THREAD

**Issue by dimsuz** [detekt/detekt#4110]
## Expected Behavior
Detekt runs with no errors

## Observed Behavior
When running a `detektDebug` task on android project which uses `kotlin-parcelize` plugin, task fails with exception:
```
error: object 'YourParcelable' is not abstract and does not implement
abstract base class member public abstract fun describeContents(): Int defined in ru.example.YourParcelable
```

## Steps to Reproduce
1. Have a module with `apply plugin: "kotlin-parcelize" and the following code

```kotlin
abstract class ScreenKey : Parcelable {
  val name: String
}
@Parcelize
object YourParcelable : ScreenKey() {
  override val name = "foo"
}
```
2. Run `detektDebug` task (this enables type resolution)
3. Observe the following error:
```
error: object 'YourParcelable' is not abstract and does not implement 
abstract base class member public abstract fun describeContents(): Int defined in ru.example.ScreenKey
```

## Context
Parcelize plugin inserts this `describeContent` function (and others) during compilation. I guess `detekt` is being hooked up into a compiler toolchain prior to this happening.

I could maybe try to come up with the PR to fix this, but I need directions: very new to this type resolution feature.

## Your Environment

* Version of detekt used: 1.18.1
* Version of Gradle used (if applicable): 7.0.2
* Operating System and version: Arch Linux

**Comment by dimsuz** [comment]
Oh, wait, this is a non-fatal error. Task is actually successful. Not as critical then, but perhaps should still be fixed.

**Comment by cortinico** [comment]
It's indeed a non fatal. Those are outputs of the compilation but Detekt is ignoring them and running correctly (could not resolve all the types). I believe this will be solved or at least mitigated by #4106

**Comment by BraisGabin** [comment]
Closing. There are other issue tracking this already. For example: #3684
